ibis/backends/pandas/client.py

```diff
@@ -192,37 +195,131 @@ def convert_any_to_timestamp(_, out_dtype, column):
             return column.dt.tz_localize(timezone)
 
 
-@sch.convert.register(object, dt.Struct, pd.Series)
-def convert_struct_to_dict(_, out_dtype, column):
-    def convert_element(values, names=out_dtype.names):
-        if values is None or isinstance(values, dict) or pd.isna(values):
-            return values
-        return dict(zip(names, values))
+@sch.convert.register(object, dt.DataType, (type(None), type(pd.NA)))
+@sch.convert.register(object, dt.Boolean, bool)
+@sch.convert.register(object, dt.String, str)
+@sch.convert.register(object, dt.Integer, (int, np.integer))
+@sch.convert.register(object, dt.Floating, float)
+@sch.convert.register(object, dt.Date, datetime.date)
+@sch.convert.register(object, dt.Timestamp, datetime.datetime)
+@sch.convert.register(object, dt.Time, datetime.time)
+@sch.convert.register(object, dt.Interval, datetime.timedelta)
+@sch.convert.register(object, dt.JSON, (list, dict, int, float))
+def scalar_no_op(_, out_dtype, v):
```

Comment:
It'd be nice if all these `sch.convert.register` calls were somewhere outside the pandas backend, since they'll be needed for any `expr.to_pandas()`/`expr.execute()` call (not just in the pandas backend). This doesn't have be done here though.